### PR TITLE
fix saturation if max is not 100

### PR DESCRIPTION
**Pull request by Garfonso**
max handling also added for hue, but should always be 360. 

max handling for saturation was partly in place, but had some bugs. 
A guy on the forum has a light with saturation max 255 and we fixed the issues he saw together, he confirmed it working

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -1359,7 +1359,7 @@ class WebServer {
                             reject(err);
                         } else {
                             if (attr_Sat) {
-                                this.adapter.setForeignState(attr_Sat.setId, s, false, {user}, err => err ? reject(err) : resolve());
+                                this.adapter.setForeignState(attr_Sat.setId, s / 100 * attr_Sat.max, false, {user}, err => err ? reject(err) : resolve());
```
Comment (GermanBluefox) [inline lib/server.js]: Does `attr_Sat.min` exists too?

File: lib/server.js [hunk 2]
```diff
@@ -1565,7 +1565,7 @@ class WebServer {
                 max: objects[hue.id].common.max || 360,
                 getParser: (entity, attr, state) => {
                     state = state || {val: 0};
-                    entity.attributes.hs_color[0] = state.val;
+                    entity.attributes.hs_color[0] = state.val / attr.max * 360;
```
Comment (GermanBluefox) [inline lib/server.js]: Should it be `state.val / attr.max * objects[hue.id].common.max` ?

File: lib/server.js [hunk 3]
```diff
@@ -1592,7 +1592,7 @@ class WebServer {
                 max: objects[sat.id].common.max || 100,
                 getParser: (entity, attr, state) => {
                     state = state || {val: 0};
-                    entity.attributes.hs_color[1] = state.val;
+                    entity.attributes.hs_color[1] = state.val / attr.max * 100;
```
Comment (GermanBluefox) [inline lib/server.js]: `state.val / attr.max * objects[sat.id].common.max` ?

**Comment by GermanBluefox** [comment]
Thank you.

File: lib/server.js (hunk 1 above)
Comment (Garfonso) [reply lib/server.js]: No, currently not. It is set here: https://github.com/Garfonso/ioBroker.lovelace/blob/hue/lib/server.js#L1592 
We could add attr_Sat.min, too. But usually it is 0 = white/black (i.e. no color at all). Other values as minimum do not make much sense. Max is a different thing and is usually 100 if sat is a percentage or 255 as byte value.

**Pull request merged**